ResetPasswordBundle/src/PasswordResetHelper.php

```diff
@@ -76,67 +109,67 @@ public function generateResetToken(object $user): PasswordResetToken
         );
     }
 
-    public function validateTokenAndFetchUser(string $fullToken): object
+    /**
+     * Validate a PasswordResetRequest and fetch user from database
+     *
+     * @param string $fullToken selector + non-hashed verifier token
+     * @return UserInterface
+     * @throws ExpiredResetPasswordTokenException
+     * @throws InvalidResetPasswordTokenException
+     * @throws \Throwable @TODO Refactor TokenGenerator and change exception signature
+     */
+    public function validateTokenAndFetchUser(string $fullToken): UserInterface
     {
-        $resetToken = $this->findToken($fullToken);
+        /** @var PasswordResetRequestInterface $resetToken */
+        $resetRequest = $this->findToken($fullToken);
 
-        if ($resetToken->isExpired()) {
+        if ($resetRequest->isExpired()) {
             throw new ExpiredResetPasswordTokenException();
         }
 
+        /** @var UserInterface $user */
```

Comment:
This won't be needed - as it should just be any object. And below, instead of `$user->getId()`, you'll use the repository's `getUserIdentifier()`

Reply:
done